From: David Rosca <nowrep@gmail.com>
Date: Mon, 21 Dec 2015 13:58:14 +0000
Subject: Containment: Don't emit appletCreated with null applet
X-Git-Url: http://quickgit.kde.org/?p=plasma-framework.git&a=commitdiff&h=bca8e87531a3cc964b415755f50abebecef68a86
---
Containment: Don't emit appletCreated with null applet

ContainmentPrivate::createApplet may return null pointer
(eg. when desktop is immutable), so check the pointer before
emitting appletCreated.

REVIEW: 126449
BUG: 356428
---


--- a/src/plasma/containment.cpp
+++ b/src/plasma/containment.cpp
@@ -364,7 +364,9 @@
 Applet *Containment::createApplet(const QString &name, const QVariantList &args)
 {
     Plasma::Applet *applet = d->createApplet(name, args);
-    emit appletCreated(applet);
+    if (applet) {
+        emit appletCreated(applet);
+    }
     return applet;
 }
 


